### PR TITLE
feature design for queue resource reservation

### PR DESCRIPTION
**Pull request by Thor-wl**
related issue:https://github.com/volcano-sh/volcano/issues/1101
Signed-off-by: Thor <1187526662@qq.com>

### CODE REVIEW

File: docs/design/queue-resource-reservation-design.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+Sort all nodes according total resource amount of one or some dimensions. Select the least nodes which rank first and 
+satisfy requirement. It's the most efficient way for preemption reservation.  
+
+* Idle First
+
+Sort all nodes according idle resource of one or some dimensions. Select the least nodes which rank first as locked nodes.
+It's suitable for non-preemption. It's should to be noted that the locked nodes are often optimal solution but suboptimal
+solution sometimes. 
+
```
Comment (Thor-wl) [inline docs/design/queue-resource-reservation-design.md:98]: add another algorithm:  sort by existing pods belonging to target queue in descending order. select nodes which can meet the reservation requirement.

**Comment by Thor-wl** [comment]
@hudson741 will finish the `implementation` section

File: docs/design/queue-resource-reservation-design.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+kind: Queue
+metadata:
+  name: q1
+spec:
+  reclaimable: true
+  weight: 1
+
+  guarantee:            // reservation key word
+    policy: Best-Effort // preemption reservation or non-preemption reservation
+    percentage: 0.5     // locked nodes number percentage in cluster
```
Comment (hudson741) [inline docs/design/queue-resource-reservation-design.md:51]: percentage:      // locked cluster total resource percentage at all dimensions
      cpu: 0.5
      memory: 0.4
      nvidia.com/gpu: 0.2

[the remark this answers is not in the document]
Comment (hudson741) [reply docs/design/queue-resource-reservation-design.md:51]: percentage:      // locked cluster total resource percentage at all dimensions
      cpu: 0.5          // locked cluster total cpu percentage
      memory: 0.4   // locked cluster total memory percentage
      nvidia.com/gpu: 0.2   // locked cluster total  nvidia.com/gpu percentage

[the remark this answers is not in the document]
Comment (Thor-wl) [reply docs/design/queue-resource-reservation-design.md:51]: > ```
> percentage:      // locked cluster total resource percentage at all dimensions
>   cpu: 0.5          // locked cluster total cpu percentage
>   memory: 0.4   // locked cluster total memory percentage
>   nvidia.com/gpu: 0.2   // locked cluster total  nvidia.com/gpu percentage
> ```

No, example like this:
```
percentage:      
  dimensions: ["cpu", "memory, "gpu", "other-scalable-resource-type"...],   
  value: 0.2  
```

**Pull request merged**